base/src/main/java/org/aya/core/repr/AyaShape.java

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) 2020-2022 Yinsen (Tesla) Zhang.
+// Use of this source code is governed by the MIT license that can be found in the LICENSE.md file.
+package org.aya.core.repr;
+
+import kala.collection.immutable.ImmutableSeq;
+import kala.collection.mutable.MutableLinkedHashMap;
+import kala.collection.mutable.MutableList;
+import kala.collection.mutable.MutableMap;
+import org.aya.core.def.Def;
+import org.aya.core.term.Term;
+import org.jetbrains.annotations.NotNull;
+
+import static org.aya.core.repr.CodeShape.CtorShape;
+import static org.aya.core.repr.CodeShape.DataShape;
+
+/**
+ * @author kiva
+ */
+public sealed interface AyaShape {
+  @NotNull CodeShape codeShape();
+  @NotNull Term transform(@NotNull Term term, @NotNull Term type);
```

Comment:
It's just the `constructorForm`

Reply:
Because this file was changed in almost every commit, I just put the function here to resolve conflicts during rebasing. It is deleted and moved to other generalized classes in #400